v0.5/fastn-lang/src/f-script/README.md

```diff
@@ -0,0 +1,221 @@
+
+Here's another example:
+
+```ftd
+-- greeting: {
+    let names = ["Bob", "Alice"]
+    let result = ""
+    for name in names {
+        result = result + " " + name
+    }
+    result
+}
+
+-- component greeting:
+string msg:
+
+-- ftd.text: $greeting.msg
+
+-- end: component
+```
+
+The block contains a list of expressions. `let` is used to bind the result of
+some expression to a variable.
+
+The value of `result` is returned since it comes last in the list of
+expressions.
+
+# Features
+
+## From `fastn 0.4`
+
+- operators (see `fastn-grammar/src/evalexpr/operator/display.rs` for a list)
+- Multiple expressions in body. The parser is able to parse multiple expression
+  in a function body but, only the first expression is evaluated. For:
+
+  ```ftd
+  -- void handle(name, email):
+  ftd.string-field name:
+  ftd.string-field email:
+
+  console.log(email.value);
+  console.log("hello"); ;; this never evaluates!
+  ```
+
+  The generated js is:
+
+  ```js
+  let test__handle = function (args)
+  {
+    let __fastn_super_package_name__ = __fastn_package_name__;
+    __fastn_package_name__ = "test";
+    try {
+      let __args__ = fastn_utils.getArgs({
+      }, args);
+      return (console.log(__args__.email.value));
+      return (console.log("hello")); // THIS WILL NEVER BE EVALUATED!
+    } finally {
+      __fastn_package_name__ = __fastn_super_package_name__;
+    }
+  }
+  ```
+
+And that's it. Anything that the parser is not able to parse/identify is simply
+converted to js if possible. Like the `console.log` call above. So it's mostly
+safe to assume that whatever js you can write in one line is valid `f-script`
+in 0.4.
+
+Exceptions to above statement include resolving variables that are defined in
+`fastn` and global variables. Global variables can be used like this:
+
+```ftd
+-- string some: someday ;; a global variable
+
+-- void handle(name, email):
+ftd.string-field name:
+ftd.string-field email:
+string x: $some ;; can only be accessed through `x` in `handle()`
+
+console.log(x);
+```
+
+Declaring `x` like this will not be necessary in 0.5. Users will simply be able
+to refer `some` that is defined outside of `handle`.
+
+## Motivation behind proposed new features
+
+The motivation to change f-script originates from the requirement that we want
+to support multiple targets (desktop/mobile/TUI etc). To do this, f-script has
+to become a base language that `fastn` will translate to:
+
+- js for the web
+- swift for ios/macos
+- C# for Windows
+- etcetera
+
+Most of the interesting stuff happens in p-script, like registering events
+(`$on-click$`). f-script is a simple procedural language that is mostly
+insipired from `rust`.
+
+## New in `fastn 0.5`
+
+- `let` bindings
+
+```ftd
+{
+  let name = "Siddhant";
+  let adj = "";
+
+  name + (adj || " (Programmer)")
+}
+```
+
+- Control statements (`if..else`, `for`, `case`)
+
+```ftd
+{
+  let res = if name == "" { ;; nested f-script block
```

Comment:
Instead of `f-script block`, let's use the terminology `block expression`, a la https://doc.rust-lang.org/beta/reference/expressions/block-expr.html (if is also f-script if, and let is also f-script let, there is no benefit of prefixing `f-script` to everything, and you are only prefixing it to `block`, making it seem like block is something special.

Reply:
This is done in https://github.com/fastn-stack/fastn/pull/2010/commits/2ab92c8f526a35437e6b0e5e55709bc1fb1d54d1